Algorithms/Dynamic-Programming/Longest Increasing Subsequence.cpp

```diff
@@ -0,0 +1,79 @@
+// This program solves the problem of Longest Increasing Subsequence(LIS)
+// Problem Statement: The Longest Increasing Subsequence (LIS) problem is to find the length of the longest 
+//                    subsequence of a given sequence such that all elements of the subsequence are sorted 
+//                    in increasing order.
+// Input Format : The first line contains n,the number of elements of the array.
+//                The second line contains n space separated integers.
+// Output Format : A single number specifying the length of LIS, followed by each element of LIS.
+
+// Approach Used : Dynamic Programming
+
+// Intuition behind dynamic programming:
+// In this solution each element of the aux vector ,say at index i stores the LIS possible with the 
+// last element arr[i](ith element of the input array). So clearly, the first element of aux stores just arr[0].
+// Next for finding LIS for other elements as last element of that sequence, an iteration is made for
+// all the previous LIS whose last element is less than the current element. If such a sequence is found
+// the largest of these is taken with the last element appended at the end of it. Finally the sixe of the maximum sequence
+// along with the sequence is printed. The memorisation of previous LIS is a key to dynamic programming here and forms the
+// overlapping subproblems.
```

Comment:
Memoization doesn't form overlapping subproblems but actually helps solve the overlapping subproblems by just solving one and remembering it so that whenever needed again we can directly return the answer.